### PR TITLE
Added focal loss

### PR DESCRIPTION
**Pull request by MarkoKostiv**

### CODE REVIEW

File: models/detr.py [hunk 1]
```diff
@@ -101,11 +101,46 @@ def __init__(self, num_classes, matcher, weight_dict, eos_coef, losses):
...
+
+    def loss_labels_focal(self, outputs, targets, indices, num_boxes, log=True):
+        """Classification loss (NLL)
+        targets dicts must contain the key "labels" containing a tensor of dim [nb_target_boxes]
+        """
+        assert 'pred_logits' in outputs
+        src_logits = outputs['pred_logits']
+
+        idx = self._get_src_permutation_idx(indices)
+        target_classes_o = torch.cat([t["labels"][J] for t, (_, J) in zip(targets, indices)])
```
Comment (awlego) [inline models/detr.py:125]: what do t and J stand for here? I recommend using non-single letter variable names, even if temporary.

**Review by awlego — APPROVED** [review]
Looks good to me. Minor readability suggestion.

File: models/detr.py (hunk 1 above)
Comment (MarkoKostiv) [reply models/detr.py:125]: I agree, this is just the original part from `detr`
